src/modules/logger/log_writer_file.cpp

```diff
@@ -42,9 +42,10 @@
 #include <px4_platform_common/posix.h>
 #include <px4_platform_common/crypto.h>
 #include <px4_platform_common/log.h>
-#ifdef __PX4_NUTTX
+#include <px4_platform_common/px4_config.h>
```

Comment:
Do we want this dependency

Reply:
Yup because of this function call.
https://github.com/PX4/PX4-Autopilot/blob/77f77a73d15be65e767e0ac4b018c74f176b9f7c/src/modules/logger/log_writer_file.cpp#L649
Before it got the declaration though a nested hardfault_log.h include, but this patch guards that.